dts/bindings/mtd/jedec,spi-nor.yaml

```diff
@@ -24,17 +24,17 @@ properties:
 
   erase-block-size:
    type: int
+   category: required
    description: address alignment required by flash erase operations
-   category: optional
 
   write-block-size:
    type: int
+   category: required
    description: address alignment required by flash write operations
-   category: optional
 
   size:
     type: int
-    category: optional
+    category: required
```

Comment:
If we are making 'size' required than we need to update all the dts files that have a "jedec,spi-nor" to have size.

Reply:
I tend to agree, but I can't do that because I don't know the size of the various flashes (or the correct erase or byte size for each).  In Linux the jedec ID is used to look up the size from a registry at runtime.

FWIW, since the driver won't build without that define, it's probably not a problem (witness that the build passed).

Reply:
Will be a problem with the new DT parser we're about to integrate, because it will flag it as an error. We spent the last few days fixing `category: required` properties.

What you could do is keep it as `category: optional`, and add a comment explaining that it ought to be required, and why it isn't at the moment.

Reply:
I'm just going to remove the patch.  The driver won't build without them, but that's nothing new, and adding a comment in the binding yaml won't help with that problem.